src/components/InfiniteHits/__tests__/InfiniteHits-test.js

```diff
@@ -114,5 +122,71 @@ describe('InfiniteHits', () => {
 
       expect(tree).toMatchSnapshot();
     });
+
+    it('should render <InfiniteHits /> with "Show previous" button on first page', () => {
+      const hits = [
+        {
+          objectID: 'one',
+          foo: 'bar',
+        },
+        {
+          objectID: 'two',
+          foo: 'baz',
+        },
+      ];
+
+      const props = {
+        hasShowPrevious: true,
+        results: { hits },
+        hits,
+        isFirstPage: true,
+        isLastPage: false,
+        templateProps: {
+          templates: {
+            item: 'item',
+            showMoreText: 'showMoreText',
+            showPreviousText: 'showPreviousText',
+          },
+        },
+        cssClasses,
+      };
+
+      const tree = mount(<InfiniteHits {...props} />);
+
+      expect(tree).toMatchSnapshot();
```

Comment:
On top of the snapshot, we could add the expectations that are specific to this test. This would avoid missing errors like we have now about the CSS classes. For instance, check that the previous button is added, that it's disabled, that the CSS classes are here, etc. What do you think?

Reply:
Yeah. I copied the previous test and adapted it for the previous button but maybe that's not the more effective way to test it.

What do you think of this?

```js
      // Test that we have two buttons (previous and next)
      expect(tree.find('button').length).toEqual(2);

      // Test that the first one (previous) is disabled since we already are on the first page
      expect(
        tree
          .find('button')
          .first()
          .props().disabled
      ).toBeDefined();

      // Test that the first one (previous) has the correct CSS classes
      expect(
        tree
          .find('button')
          .first()
          .hasClass('loadPrevious disabledLoadPrevious')
      ).toEqual(true);
```

(but in that case I think the snapshot test is no longer useful and could be removed)

Reply:
We should keep the `.loadMore` button apart from this test, so I wouldn't test that there are two button there.

This "jQuery-like" way of testing is very fragile IMO because it would break easily if we were to change the order of the generated DOM, and we would lose the intent of the test again.

We could test it more declaratively and keep the snapshot:

- expect that `.loadPrevious` exists
- expect that it has the `.disabledLoadPrevious` class
- expect that it has the disabled prop

That would be my guess. I might be wrong though, let me know what you think.